src/essence/Tools/Measure/MeasureTool.js

```diff
@@ -764,13 +792,23 @@ function makeGhostLine(lng, lat) {
         //distMousePoint.bindTooltip("" + roundedTotalDist + "m\n (+" + roundedDist + "m) " + distAzimuth + "&deg;",
         //  {permanent: true, direction: 'right', className: "distLabel", className: "noPointerEvents", offset: [15,-15]})
         //distMousePoint.addTo(Map_.map);
-        CursorInfo.update(
-            `${roundedTotalDist}m ${
-                mode === 'continuous' ? `(+${roundedDist}m)` : ''
-            } ${distAzimuth}&deg;`,
-            null,
-            false
-        )
+        if(distDisplayUnit == 'meters'){
+            CursorInfo.update(
+                `${roundedTotalDist}m ${
+                    mode === 'continuous' ? `(+${roundedDist}m)` : ''
+                } ${distAzimuth}&deg;`,
+                null,
+                false
+            )
+        }else if (distDisplayUnit == "kilometers"){
+            CursorInfo.update(
+                `${(roundedTotalDist/1000).toFixed(2)}km ${
+                    mode === 'continuous' ? `(+${roundedDist/1000}km)` : ''

```

Comment:
I think you also want to have `toFixed(2)` here for consistency.

Reply:
Oh I totally missed that, will fix that.